pkg/image/prune/prune.go

```diff
@@ -240,28 +240,41 @@ func NewPruner(options PrunerOptions) (Pruner, error) {
 	}
 	algorithm.namespace = options.Namespace
 
-	g := graph.New()
-	addImagesToGraph(g, options.Images, algorithm)
-	addImageStreamsToGraph(g, options.Streams, options.LimitRanges, algorithm)
-	addPodsToGraph(g, options.Pods, algorithm)
-	addReplicationControllersToGraph(g, options.RCs)
-	if err := addBuildConfigsToGraph(g, options.BCs); err != nil {
-		return nil, err
+	p := &pruner{
+		algorithm:      algorithm,
+		registryClient: options.RegistryClient,
+		registryURL:    options.RegistryURL,
 	}
-	if err := addBuildsToGraph(g, options.Builds); err != nil {
+
+	if err := p.buildGraph(options); err != nil {
 		return nil, err
 	}
-	addDaemonSetsToGraph(g, options.DSs)
-	addDeploymentsToGraph(g, options.Deployments)
-	addDeploymentConfigsToGraph(g, options.DCs)
-	addReplicaSetsToGraph(g, options.RSs)
 
-	return &pruner{
-		g:              g,
-		algorithm:      algorithm,
-		registryClient: options.RegistryClient,
-		registryURL:    options.RegistryURL,
-	}, nil
+	return p, nil
+}
+
+// buildGraph builds a graph
+func (p *pruner) buildGraph(options PrunerOptions) error {
+	p.g = graph.New()
+
+	p.addImagesToGraph(options.Images, p.algorithm)
```

Comment:
p.algorithm is a part of `pruner` and can be removed from arguments.

Reply:
> p.algorithm is a part of pruner and can be removed from arguments.

Removed.